### PR TITLE
Version 5: overall revamp (see details)

### PR DESCRIPTION
**Pull request by albertodebortoli**
This PR addresses the following:

- [x] Overall renaming
- [x] Overall refactoring and simplification
- [x] Overall cleanup
- [x] Move `EphemeralConfiguration` from demo app to framework
- [x] Harmonise header comments
- [x] Inverse the configuration priority order (from highest to lowest)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,134 +36,134 @@ pod "JustTweak"
...
-    // tweak was found in some configuration, use tweak.value
+// tweak was found in some configuration, use tweak.value
 } else {
-    // tweak was not found in any configuration
+// tweak was not found in any configuration
 }
 ```
 
-`@FeatureFlag` and `@FeatureFlagWrappingOptional` are available to mark properties representing feature flags. Mind that by using these property wrappers, a static instance of `TweaksConfigurationsCoordinator` must be available.
+`@FeatureFlag` and `@OptionalTweak` are available to mark properties representing feature flags. Mind that by using these property wrappers, a static instance of `JustTweak` must be available.
```
Comment (antoniostrijdom) [inline README.md]: I think you mean `@TweakProperty`?

Comment (albertodebortoli) [reply README.md]: I was just updating the README when you posted the comment :)

File: JustTweak/Classes/Configurations/UserDefaultsConfiguration.swift [hunk 2]
```diff
@@ -69,4 +67,15 @@ final public class UserDefaultsTweaksConfiguration: NSObject, MutableTweaksConfi
         }
         return nil
     }
+        
+    private func updateUserDefaults(value: TweakValue, feature: String, variable: String) {
+        userDefaults.set(value, forKey: userDefaultsKeyForTweakWithIdentifier(variable))
+        userDefaults.synchronize()
```
Comment (dchakarov) [inline JustTweak/Classes/Configurations/UserDefaultsConfiguration.swift]: Do we still need this?

File: JustTweak/Classes/UI/TweakViewController.swift [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
     var desc: String? { get set }
     var value: TweakValue { get set }
-    var delegate: TweaksConfigurationViewControllerCellDelegate? { get set }
+    var delegate: TweakViewControllerCellDelegate? { get set }
 }
 
-internal protocol TweaksConfigurationViewControllerCellDelegate: class {
-    func tweaksConfigurationCellDidChangeValue(_ cell: TweaksConfigurationViewControllerCell)
+internal protocol TweakViewControllerCellDelegate: class {
+    func tweaksConfigurationCellDidChangeValue(_ cell: TweakViewControllerCell)
```
Comment (dchakarov) [inline JustTweak/Classes/UI/TweakViewController.swift]:
```suggestion
    func tweakConfigurationCellDidChangeValue(_ cell: TweakViewControllerCell)
```

File: JustTweak/Classes/UI/TweakViewController.swift [hunk 4]
```diff
@@ -238,29 +225,28 @@ extension TweaksConfigurationViewController {
     }
 }
 
-extension TweaksConfigurationViewController: TweaksConfigurationViewControllerCellDelegate {
+extension TweakViewController: TweakViewControllerCellDelegate {
     
-    internal func tweaksConfigurationCellDidChangeValue(_ cell: TweaksConfigurationViewControllerCell) {
+    internal func tweaksConfigurationCellDidChangeValue(_ cell: TweakViewControllerCell) {
```
Comment (dchakarov) [inline JustTweak/Classes/UI/TweakViewController.swift]:
```suggestion
    internal func tweakConfigurationCellDidChangeValue(_ cell: TweakViewControllerCell) {
```

File: JustTweak/Classes/Configurations/UserDefaultsConfiguration.swift (hunk 2 above)
Comment (albertodebortoli) [reply JustTweak/Classes/Configurations/UserDefaultsConfiguration.swift]: nope :)

**Pull request merged**